src/docker-osx-dev

```diff
@@ -1031,6 +1034,23 @@ function install {
 }
 
 #
+# Fetch the latest docker-osx-dev script, and then run install
+#
+function update_self {
+  log_info "Updating to the latest docker-osx-dev"
+  local readonly current_file=$(which docker-osx-dev)
```

Comment:
I am not familiar with BASH_SOURCE[0].  I tested on (bash 3.x and bash 4.x)

```

echo "1: {$BASH_SOURCE}"
echo "2: {$BASH_SOURCE[0]}"
```

And received

```
1: {/tmp/aha}
2: {/tmp/aha[0]}
```